bundled/boost-1.62.0/include/boost/mpl/assert.hpp

```diff
@@ -185,14 +185,12 @@ template< typename P > struct assert_arg_pred_not
 };
 
 template< typename Pred >
-failed ************ (Pred::************ 
-      assert_arg( void (*)(Pred), typename assert_arg_pred<Pred>::type )
-    );
+failed ************ Pred::************
```

Comment:
I'm really frightened to see that this is even possible in C++.

Reply:
This is a really remarkably clever way to print an error message, I think.

Reply:
How bizarre. Who comes up with this!